web/src/interactives/covid-dashboard/stategraph.js

```diff
@@ -34,81 +36,84 @@ class StateGraph extends Component {
     this.loadData(this.state.state);
   }
 
-  async componentDidUpdate(prevProps, prevState, snapshot) {
+  componentDidUpdate(prevProps, prevState, snapshot) {
     if (prevState.state !== this.state.state) {
       this.loadData(this.state.state);
     }
   }
 
-  async loadData(state) {
-    const data = await fetchStateData(state);
-    this.setState({
-      chartData: {
-        labels: data.dates,
-        datasets:
-          this.state.sort === "daily"
-            ? [
-              {
-                label: "Hospitalized Currently",
-                data: Object.values(data.hospitalizedCurrently),
-                fill: true,
-                borderColor: "#455574",
-                pointRadius: 0
-              },
-              {
-                label: "Increase in Cases",
-                data: Object.values(data.positiveIncrease),
-                fill: true,
-                borderColor: "#C63F3F",
-                pointRadius: 0
-              },
-              {
-                label: "Increase in Deaths",
-                data: Object.values(data.deathIncrease),
-                fill: true,
-                borderColor: "#251616",
-                pointRadius: 0
-              },
-              {
-                label: "Moving Average of Daily Cases",
-                data: Object.values(data.movingAvgCases),
-                fill: true,
-                borderColor: "#F4B436",
-                pointRadius: 0
-              },
-            ]
-            : [
-              {
-                label: "Confirmed",
-                data: data.confirmed,
-                fill: true,
-                borderColor: "#C63F3F",
-                pointRadius: 0
-              },
-              {
-                label: "Deaths",
-                data: Object.values(data.deaths),
-                fill: true,
-                borderColor: "#251616",
-                pointRadius: 0
-              },
-              {
-                label: "Recovered",
-                data: Object.values(data.recovered),
-                fill: true,
-                borderColor: "#455574",
-                pointRadius: 0
-              },
-            ],
-      },
-    });
+  loadData(state) {
+    this.setState({ loading: true });
+    fetchStateData(state).then((data) =>
+      this.setState({
+        loading: false,
+        chartData: {
+          labels: data.dates,
+          datasets:
+            this.state.sort === "daily"
+              ? [
+                  {
+                    label: "Hospitalized Currently",
+                    data: Object.values(data.hospitalizedCurrently),
+                    fill: true,
+                    borderColor: "#455574",
+                    pointRadius: 0,
+                  },
+                  {
+                    label: "Increase in Cases",
+                    data: Object.values(data.positiveIncrease),
+                    fill: true,
+                    borderColor: "#C63F3F",
```

Comment:
Are these Theme UI colors? If so, you can import theme above (`import theme from styles/theme.js` or something) and then access `theme.colors.primary`.